src/formatDate.js

```diff
@@ -8,7 +8,66 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldYear = yearFormatAndPlace(fromFormat);
+  const newYear = yearFormatAndPlace(toFormat);
+
+  const formatedDate = [];
+  let year = '';
+
+  const dateParts = date.split(fromFormat[3]);
+  const newDateSeparator = toFormat[3];
+
+  const thresholdYear = 30;
+
+  if (oldYear.format !== newYear.format) {
+    if (oldYear.format.length < newYear.format.length) {
+      year =
+        dateParts[oldYear.place] < thresholdYear
```

Comment:
The comparison `dateParts[oldYear.place] < thresholdYear` might not work as expected because `dateParts[oldYear.place]` is a string. Implicit type coercion in JavaScript can lead to unexpected results when comparing strings with numbers. It would be safer to explicitly convert the string to a number before the comparison.